pocket_coffea/workflows/base.py

```diff
@@ -703,6 +704,31 @@ def get_shape_variations(self):
                 for jet_coll_name, jet_coll in jets_calibrated.items():
                     self.events[jet_coll_name] = jet_coll
 
+                if self.params.lepton_scale_factors.electron_sf["apply_eleSS"][self._year]:
+                    etaSC = abs(self.events["Electron"]["deltaEtaSC"] + self.events["Electron"]["eta"])
+                    self.events["Electron"] = ak.with_field(
+                        self.events["Electron"], etaSC, "etaSC"
+                    )
+                    self.events["Electron"] = ak.with_field(
+                        self.events["Electron"], self.events["Electron"]["pt"], "pt_original"
+                    )
+                    ssfile = self.params.lepton_scale_factors["electron_sf"]["JSONfiles"][self._year]["fileSS"]
+                    # Apply smearing on MC, scaling on Data
+                    if self._isMC:
+                        ele_pt_smeared = get_ele_smeared(self.events["Electron"], ssfile, self._isMC, nominal=True)
+                        self.events["Electron"] = ak.with_field(
```

Comment:
this can also be expressed with `self.events["Electron", "pt"] =..` but I don't know which one is faster/better